streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrint.java

```diff
@@ -62,13 +62,19 @@ public void init(ProcessorContext context) {
             if (valueSerde == null) {
                 this.valueSerde = context.valueSerde();
             }
-        }
 
-        @Override
-        public void process(final K key, final V value) {
-            final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
-            final V deValue = (V) maybeDeserialize(value, valueSerde.deserializer());
-            action.apply(deKey, deValue);
+            // no mapper provided, using a default one which takes care of checking byte[] for deserialization
+            if (mapper == null) {
+                mapper = new KeyValueMapper<K, V, String>() {
+                    @Override
+                    public String apply(K key, V value) {
+                        final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
```

Comment:
I'm a little bit confused here because, if we do something like that in the default mapper : 

`byte[] keyBytes = ((Serde<K>)keySerde).serializer().serialize(null, key);`
`byte[] valueBytes = ((Serde<V>)valueSerde).serializer().serialize(null, value);`

Then I see two problems : 
- how to get "meaningful" strings from there (i.e. if value was a String, then to bytes, then using new String(bytes) gives me the original string but i.e. if key was an Integer ?)
- if the key/value are already byte[ ] but for i.e. the type inference from the KStreamPrint is <Integer, String>, then the default mapper will be in this way so i.e. key serde will be a Serde<Integere> .. so it won't work on byte[ ]

Can you elaborate what are you trying to achieve here please ? For sure, I'm missing something ...